### PR TITLE
removes the remainder from ScalerFor method

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**:
this PR removes existing scalers from `ScalerFor` method

**Release note**:

```release-note
`kubectl scale` can now scale any resource (kube, CRD, aggregate) conforming to the standard scale endpoint
```

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
An intermediary step that removes existing scalers from `ScalerFor` method. This is just to narrow the scope in case of potential issues.

File: pkg/kubectl/scale.go [hunk 1]
```diff
@@ -167,6 +153,7 @@ func (precondition *ScalePrecondition) ValidateReplicationController(controller
 	return nil
 }
 
+// TODO(p0lyn0mial): remove ReplicationControllerScaler
```
Comment (deads2k) [inline pkg/kubectl/scale.go]: Being held for the reapers?  If so, open an issue.

File: pkg/kubectl/scale.go [hunk 2]
```diff
@@ -54,21 +53,8 @@ type Scaler interface {
...
-	case extensions.Kind("ReplicaSet"), apps.Kind("ReplicaSet"):
-		return &ReplicaSetScaler{c.Extensions()}, nil
-	case batch.Kind("Job"):
-		return &JobScaler{c.Batch()}, nil // Either kind of job can be scaled with Batch interface.
-	case apps.Kind("StatefulSet"):
-		return &StatefulSetScaler{c.Apps()}, nil
-	default:
-		return &GenericScaler{scalesGetter, gr}, nil
-	}
+func ScalerFor(scalesGetter scaleclient.ScalesGetter, gr schema.GroupResource) Scaler {
```
Comment (deads2k) [inline pkg/kubectl/scale.go]: Function is no longer needed.

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -2683,16 +2683,12 @@ func RemoveAvoidPodsOffNode(c clientset.Interface, nodeName string) {
 	ExpectNoError(err)
 }
 
-//TODO(p0lyn0mial): remove internalClientset and kind
-func getScalerForKind(internalClientset internalclientset.Interface, kind schema.GroupKind, scalesGetter scaleclient.ScalesGetter, gr schema.GroupResource) (kubectl.Scaler, error) {
-	return kubectl.ScalerFor(kind, internalClientset, scalesGetter, gr)
+func getScalerForKind(scalesGetter scaleclient.ScalesGetter, gr schema.GroupResource) kubectl.Scaler {
```
Comment (deads2k) [inline test/e2e/framework/util.go]: Function is no longer needed

**Comment by deads2k** [comment]
@soltysh What's up with this error: 

`error: Scaling the resource failed with: could not fetch the scale for jobs.batch pi: the server could not find the requested resource`

**Comment by p0lyn0mial** [comment]
> What's up with this error:
>
> error: Scaling the resource failed with: could not fetch the scale for jobs.batch pi: the server could > not find the requested resource

Does a Job resource have a Scale subresource ?

**Comment by deads2k** [comment]
> What's up with this error:
> 
> error: Scaling the resource failed with: could not fetch the scale for jobs.batch pi: the server could > not find the requested resource
> Does a Job resource have a Scale subresource ?

No it doesn't.  @soltysh naughty naughty naughty.  You'll need to add that.

File: pkg/kubectl/scale.go (hunk 1 above)
Comment (p0lyn0mial) [reply pkg/kubectl/scale.go]: Yes, that is correct. I was planning to deal with the reapers in the very next pull. In that case do we need an issue for that ?

Unless your plan is to leave the reapers for the time being and focus on preparing kubectl to support scaling custom resources.

**Comment by deads2k** [comment]
@p0lyn0mial it's going to take me a little while to get the jobs endpoint sorted out.  Can you replace the others?

**Comment by p0lyn0mial** [comment]
commits squashed, vet fixed.
we have to wait for @soltysh  pull to be merged before we can squash that one in.

**Comment by deads2k** [comment]
gofmt

**Comment by p0lyn0mial** [comment]
> gofmt

believe or not but I ran `hack/update-all` :)

**Comment by Random-Liu** [comment]
Also https://github.com/kubernetes/kubernetes/issues/59058.

Agree with reverting this.

**Pull request merged**